src/ProjectTemplates/test/RazorComponentsTemplateTest.cs

```diff
@@ -75,35 +75,31 @@ private void TestBasicNavigation()
             // Give components.server enough time to load so that it can replace
             // the prerendered content before we start making assertions.
             Thread.Sleep(5000);
-            Browser.WaitForElement("ul");
+            Browser.Exists(By.TagName("ul"));
             // <title> element gets project ID injected into it during template execution
-            Assert.Contains(Project.ProjectGuid, Browser.Title);
+            Browser.Equal(Project.ProjectName.Trim(), () => Browser.Title.Trim());
 
             // Initially displays the home page
-            Assert.Equal("Hello, world!", Browser.GetText("h1"));
+            Browser.Equal("Hello, world!", () => Browser.FindElement(By.TagName("h1")).Text);
 
             // Can navigate to the counter page
-            Browser.Click(By.PartialLinkText("Counter"));
-            Browser.WaitForUrl("counter");
```

Comment:
@javiercn Just so we can understand what the right APIs are to use in the future, could you clarify why `WaitForUrl` is wrong? Does it not wait in the right way or something?

Reply:
It's mostly that it was in a separate class with a different way to wait and with different settings to configure it. I want us to share the asserts we use for E2E testing so that when we make an improvement it translates to the rest of the codebase.

Reply:
Ah, so `WaitForUrl` is our own custom thing and not something built-in? If so then I totally understand. Yes it definitely makes sense to have only one set of custom wait-assertions!

Reply:
Yeah, all these methods were helpers on the templating test project. The ones in blazor are better IMO. If we need more, they should be added there.